catalyst-gateway/bin/src/cardano/mod.rs

```diff
@@ -295,6 +268,131 @@ async fn index_block(
     };
 }
 
+///
+async fn index_blocks(
+    db: &EventDB, genesis_values: &GenesisValues, network_str: &str, blocks: &[MultiEraBlock<'_>],
+) -> bool {
+    let values: Vec<_> = blocks
+        .iter()
+        .filter_map(|block| {
+            IndexedFollowerDataParams::from_block_data(genesis_values, network_str, block)
+        })
+        .collect();
+
+    match db.index_many_follower_data(&values).await {
+        Ok(()) => {
+            info!(count = values.len(), "Finished indexing block data");
+            true
+        },
+        Err(e) => {
+            error!(error = ?e, "Failed to write DB entries");
+            false
+        },
+    }
+}
+
+/// Index transactions (and its inputs and outputs) from a slice of blocks.
+async fn index_transactions(db: &EventDB, blocks: &[MultiEraBlock<'_>], network_str: &str) -> bool {
+    let blocks_txs: Vec<_> = blocks
+        .iter()
+        .flat_map(|b| b.txs().into_iter().map(|tx| (b.slot(), tx)))
+        .collect();
+
+    // Index transaction data
```

Comment:
Can we consider extracting `Index transaction data,` `Index transaction output data` and ` Index transaction input data` to functions?